src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComActivator.cs

```diff
@@ -533,10 +533,10 @@ public string RequestLicKey(Type type)
         // If we are being entered because of a call to ICF::CreateInstanceLic(),
         // "isDesignTime" will be "false" and "key" will point to a non-null
         // license key.
-        public object AllocateAndValidateLicense(Type type, string? key, bool isDesignTime)
+        public object? AllocateAndValidateLicense(Type type, string? key, bool isDesignTime)
```

Comment:
>? [](start = 21, length = 1)

Can this be null?  I'm curious because of the method name; it makes it sound like only a valid license would be returned, and I'm assuming (maybe incorrectly) that null would be an invalid license.

It doesn't matter much for this method itself, as it's internal; I just wonder whether this nullable-ness is flowing out and impacting the annotations we're adding to callers that are exposed publicly.

Reply:
> Can this be null? I'm curious because of the method name; it makes it sound like only a valid license would be returned, and I'm assuming (maybe incorrectly) that null would be an invalid license.

marked nullable as MethodBase.Invoke(...) method called on return has nullable return. You mean if validation failed it would throw instead of returning null? From try catch most likely that is true

> It doesn't matter much for this method itself, as it's internal; I just wonder whether this nullable-ness is flowing out and impacting the annotations we're adding to callers that are exposed publicly.

Checked publicly exposed method of IClassFactory and IClassFactory2 interface
```
void CreateInstance(
            [MarshalAs(UnmanagedType.Interface)] object? pUnkOuter,
            ref Guid riid,
            [MarshalAs(UnmanagedType.Interface)] out object? ppvObject);
```
Seems paramete `out object? ppvObject` should be non nullable to me

Reply:
> marked nullable as MethodBase.Invoke(...) method called on return has nullable return

That's because any method with any return value might be returned.  You'd need to examine the particular method being invoked.

> You mean if validation failed it would throw instead of returning null?

Yes, that's my question.  Maybe @AaronRobinsonMSFT knows.

> Seems paramete out object? ppvObject should be non nullable

The docs state:
```
If the object does not support the interface specified in riid, the implementation must set *ppvObject to NULL.
```
so I believe nullable here is correct.  But that doesn't mean that it might be null for every use of this method, just that some uses could result in null.

Reply:
>>You mean if validation failed it would throw instead of returning null?

>Yes, that's my question. Maybe @AaronRobinsonMSFT knows.

None of these COM functions can return `null` during allocation. Is this scenario the .NET Exception will be marshalled as an `HRESULT` since the function method is marked as returning `void`. This can be observed in the exception below that throws `CLASS_E_NOTLICENSED` which will result in an error `HRESULT` to the caller - which means by COM semantics that all out parameters are in an undefined state.

Reply:
thank you, updated only this method to return non null